src/lib/components/ui/menu/menu-item.svelte

```diff
@@ -0,0 +1,58 @@
+<script lang="ts">
+	import { getContext } from 'svelte'
+	import type { HTMLAnchorAttributes } from 'svelte/elements'
+	import type { Dimension, MenuStore } from './menu-store.svelte'
+	interface Props extends HTMLAnchorAttributes {
+		dimensions?: Dimension
+	}
+	let { dimensions, children, ...restProps }: Props = $props()
+
+	const store = getContext<MenuStore>('menu-store')
+</script>
+
+<a {...restProps} class={dimensions ?? store.size}>
+	{#if children}
+		{@render children()}
+	{/if}
+</a>
```

Comment:
This should be both `button` and `a` element. Ideally, here we should just use `Button` component we already have